### ISSUE TITLE
Handle softmax derivative

### ISSUE THREAD

**Issue by opixelum** [opixelum/atamai#14]
Since softmax gives a vector as an output, we can't define a gradient to it. The workaround is to take derivatives of the loss function, with respect to the inputs passed to softmax, instead of the outputs passed by softmax.

Reference: https://stackoverflow.com/questions/58461808/understanding-backpropagation-with-softmax/76532286#76532286
